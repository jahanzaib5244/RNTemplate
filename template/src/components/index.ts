export {default as AppButton} from './AppButton/AppButton';
export {default as AppText} from './AppText/AppText';
export {default as DatePicker} from './DatePicker/DatePicker';
export {default as AppInput} from './AppInput/AppInput';
export {default as AppAlert} from './AppAlert/AppAlert';
export {default as TimePicker} from './TimePicker/TimePicker';
export {default as AppHeader} from './AppHeader/AppHeader';
export {default as ThemeButton} from './ThemeButton/ThemeButton';
export {default as AppScreen} from './AppScreen/AppScreen';
export {default as BottomSheet} from './BottomSheet/BottomSheet';
export {default as AppDropdown} from './AppDropdown/AppDropdown';
