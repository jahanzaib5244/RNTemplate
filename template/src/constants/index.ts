export {default as DIMENSION} from './DIMENSION';
export {default as COLORS} from './COLORS';
export {default as IMAGES_PATH} from './IMAGES_PATH';
