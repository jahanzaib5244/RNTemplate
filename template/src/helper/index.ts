export * from './AppAlert';
export * from './AppAlert';
export * from './AppAlert';
export * from './RunToast';
export * from './KeyboardListner';
export * from './dimensions';
