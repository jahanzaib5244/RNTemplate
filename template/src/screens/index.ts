// home screens
export {default as Home} from './Home/Home';
export {default as Rating} from './Rating/Rating';
export {default as Chat} from './chat/Chat';

// auth screens
export {default as Login} from './Login/Login';
export {default as Signup} from './Signup/Signup';
export {default as OnBoard} from './OnBoard/OnBoard';
