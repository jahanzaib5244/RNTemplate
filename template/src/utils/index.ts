export * from './callApi';
export * from './STORE';
export * from './storage';
export * from './useSafeAreaInsetsStyle';
