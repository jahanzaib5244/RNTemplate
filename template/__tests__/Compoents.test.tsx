import { fireEvent, render,screen } from '@testing-library/react-native'
import 'react-native'
import { AppButton, AppText } from '../src/components'
import {Text} from 'react-native'
import rendered from 'react-test-renderer'

jest.mock('@shopify/react-native-skia')

describe('AppText Component',()=>{
    it('snapshot testing Apptext',async()=>{
    const tree= rendered.create(<AppText>hello</AppText>).toJSON()
    expect(tree).toMatchSnapshot();
    })

    it('with some string',async()=>{
    const { getByText} =   render(<AppText>hello world</AppText>) 
    expect(getByText('hello world').children[0]).toBe('hello world')
    })

    it('with compoment',async()=>{
    const { getAllByText} =   render(<AppText><Text>hello world</Text><Text>hello world</Text></AppText>) 
    expect(getAllByText('hello world').length).toBe(2)
    })
})

describe('App Button Component',()=>{
    it('snapshot testing button',async()=>{
        const tree= rendered.create(<AppButton title='Title' />).toJSON()
        expect(tree).toMatchSnapshot();
    })
    it('with title',async()=>{
    const { getAllByText} =   render(<AppButton title='Login' />) 
    expect(getAllByText('Login').length).toBe(1)
    })

    it('with loading true',async()=>{
    const fun =  jest.fn(()=>console.log('hello'))
    const { getByTestId } =   render(<AppButton title='Login' onPress={fun} loading={true} />) 
    fireEvent.press(getByTestId('button.presaable'))
    getByTestId('activity.indicator')
    expect(fun.mock.calls).toHaveLength(0)
    })

    it('with loading false',async()=>{
    const fun =  jest.fn(()=>console.log('hello'))
    const { getByTestId,getByText } =   render(<AppButton title='Login' onPress={fun} loading={false} />) 
    getByText('Login')
    fireEvent.press(getByTestId('button.presaable'))
    expect(fun.mock.calls).toHaveLength(1)
    })
})

