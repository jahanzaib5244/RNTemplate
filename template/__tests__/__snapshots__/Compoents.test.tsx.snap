// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Button Component snapshot testing button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#0A84FF",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": 675,
      },
      {},
    ]
  }
  testID="button.presaable"
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontSize": 15,
        },
        [
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "500",
          },
          {},
        ],
      ]
    }
  >
    Title
  </Text>
</View>
`;

exports[`AppText Component rendered 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
        "fontSize": 15,
      },
      undefined,
    ]
  }
>
  hello
</Text>
`;

exports[`AppText Component snapshot testing 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
        "fontSize": 15,
      },
      undefined,
    ]
  }
>
  hello
</Text>
`;

exports[`AppText Component snapshot testing Apptext 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
        "fontSize": 15,
      },
      undefined,
    ]
  }
>
  hello
</Text>
`;
